x-pack/plugins/enterprise_search/server/routes/app_search/search_settings.test.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MockRouter, mockRequestHandler, mockDependencies } from '../../__mocks__';
+
+import { registerSearchSettingsRoutes } from './search_settings';
+
+describe('search settings routes', () => {
+  const boosts = { foo: [{}] };
+  const resultFields = { foo: {} };
+  const searchFields = { foo: {} };
+  const searchSettings = {
+    boosts,
+    result_fields: resultFields,
+    search_fields: searchFields,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('GET /api/app_search/engines/{name}/search_settings/details', () => {
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'get',
+        path: '/api/app_search/engines/{engineName}/search_settings/details',
+      });
+
+      registerSearchSettingsRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
```

Comment:
Naive question, but why is this in a `beforeEach` rather than a `beforeAll`?  Couldn't the mock persist since it's not being changed in between `it` tests?

Reply:
I was definitely doing more than I needed to do here. `jest.clearAllMocks` was redundant as I was already doing that in a beforeAll at the root.

Instantiating `mockRouter` only ever needs to be done once, EXCEPT in the query tests, which I had to create a new router for because it tests query validation not body, and you can only validate one or the other with a mock router.

`registerSearchSettingsRoutes` must be done in a `beforeEach`, because it registers a call to a mock that is required for route specs, and since mocks get cleared before every test, it will fail every test.

https://github.com/elastic/kibana/pull/89308/commits/dcc92e3861ca266189421b438d361b9498296679